### ISSUE TITLE
Prevent the browser from caching and restoring the selected option when navigating back

### ISSUE THREAD

**Issue by renatonascalves** [alleyinteractive/elasticsearch-extensions#52]
### Description

TL:DR; For sites using `select` and `checkboxes`, there is a browser behavior which impacts those form fields. The value selected is restored, by the browser, when navigating back.

This behavior creates a very weird experience. Where it looks like the field is selected, or checked, but it is not.

This Github issue is about investigating a workable fix for this problem.

--- 

While researching a fix for this problem, I discovered a few options. But none of them seems to be the complete fix.

### Adding `autocomplete="off"` to the html field

This option seems to work sometimes, particularly for new fields.

> By setting autocomplete to "off," we are indicating to the browser that it should not remember and restore the selected option when the user navigates back to the page.

### Randomly Generate name Attributes

Not ideal in our situation since the field name is mapped to the query var.

> Some browsers may still attempt to autocomplete fields based on their name attribute. We could try generating random name attributes for our <select> elements to make them less predictable.

### Use Javascript

> Use JavaScript to clear the selected option when the page is loaded or when the user navigates back.

This option works but introduces a new caveat. The user loses the history of the selected options. If the form has multiple fields, one can navigate back and the form would indicate it is reset when the url has the params for said field.

**Comment by mboynes** [comment]
I don't think this is something that ES Extensions can (or should) address, mainly because this plugin doesn't provide any client-side JS. It is very easy to address with JS though, by resetting the form onload:

```js
window.onload = () => document.getElementById('search-form').reset();
```

This will remove any state changes made to the form fields and reset them back to their initial value. This assumes the form's initial state is set server-side (e.g. something like `<input type="text" name="foo" value="<?php echo esc_attr( get_query_var( 'foo' ) ); ?>" />`).
